src/swell/deployment/prepare_config_and_suite/prepare_config_and_suite.py

```diff
@@ -0,0 +1,601 @@
+# (C) Copyright 2021- United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+import copy
+import os
+import yaml
+
+from swell.swell_path import get_swell_path
+from swell.deployment.prepare_config_and_suite.question_and_answer_cli import GetAnswerCli
+from swell.deployment.prepare_config_and_suite.question_and_answer_defaults import GetAnswerDefaults
+from swell.utilities.jinja2 import template_string_jinja2
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+"""
+Preparation of the configuration happens in several steps. Note that the configuration determines
+templates and construction of the flow.cylc file.
+
+1. The user chooses the suite they wish to run.
+
+2. The code then builds a list of suite questions that do not depend on the model. At around the
+   same time the code determines all the tasks within that suite that do not depend on the model
+   component. At that point all these questions are asked using the question client.
+
+3. If there is some dependency on model component(s) in the flow.cylc then the code will ask the
+   user which model components should be included. Then all the suite questions depending on the
+   model will be
+
+3. All the
+"""
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+class PrepareExperimentConfigAndSuite:
+
+    # ----------------------------------------------------------------------------------------------
+
+    def __init__(self, logger, suite, platform, config_client, override, advanced):
```

Comment:
`advanced` is never used in this Class, so we should remove it.

Also, per #349, as long as we are committing to Python >3.5 (which I strongly recommend!), I suggest adding some type hints:

```suggestion
    def __init__(
        self,
        logger: Logger,
        suite: str,
        platform: str,
        config_client: str,
        override: Union[str | dict | None]
    ):
```

Also, I'm not a huge fan of `override: Union[str, dict] --- I would rather it just be `dict` and the `str` case (path to override file) be handled much earlier, when we parse command line arguments.